crates/typst-library/src/loading/toml.rs

```diff
@@ -26,12 +25,46 @@ use crate::loading::{DataSource, Load, Readable};
 /// Authors: #(details.authors
 ///   .join(", ", last: " and "))
 /// ```
+///
+/// # Conversion details { #conversion }
+///
+/// First of all, TOML documents are tables. Other values must be put in a table
+/// to be encoded or decoded.
+///
+/// | TOML value | Converted into Typst |
+/// | ---------- | -------------------- |
+/// | string     | [`str`]              |
+/// | integer    | [`int`]              |
+/// | float      | [`float`]            |
+/// | boolean    | [`bool`]             |
+/// | datetime   | [`datetime`]         |
+/// | array      | [`array`]            |
+/// | table      | [`dictionary`]       |
+///
+/// Be aware that **TOML integers** larger than 2<sup>63</sup>-1 or smaller than
+/// -2<sup>63</sup> cannot be represented losslessly in Typst, and an error will
+/// be thrown according to the [spec](https://toml.io/en/v1.0.0#integer).
+///
+/// | Typst value                           | Converted into TOML            |
+/// | ------------------------------------- | ------------------------------ |
+/// | types that can be converted from TOML | corresponding TOML value       |
+/// | `{none}`                              | ignored                        |
+/// | [`bytes`]                             | string via [`repr`]            |
+/// | [`symbol`]                            | string                         |
+/// | [`content`]                           | a table describing the content |
+/// | other types ([`length`], etc.)        | string via [`repr`]            |
+///
+/// - **Bytes** are not encoded as TOML arrays for performance reasons.
```

Comment:
Not just performance reasons, but also UX reasons as simple queries for e.g.a figure would turn into huge output. Not sure whether that's worth mentioning, but this statement could feel better motivated.